lib/search.rb

```diff
@@ -629,7 +629,12 @@ def add_more_topics_if_expected
 
   # If we're searching for a single topic
   def single_topic(id)
-    post = Post.find_by(topic_id: id, post_number: 1)
+    if @opts[:restrict_to_archetype].present?
+      topic = Topic.find_by(id: id, archetype: @opts[:restrict_to_archetype] == "regular" ? Archetype.default : Archetype.private_message)
+      post = topic&.first_post
```

Comment:
This ends up executing two queries as opposed to one. I think we can do a joins on the topics table to look for the post we need.